src/main/java/backend/RepoID.java

```diff
@@ -0,0 +1,53 @@
+package backend;
+
+import java.util.Optional;
+
+public class RepoID {
+
+    private String repoIDString;
+    private String repoOwner;
+    private String repoName;
+
+    public static Optional<RepoID> getRepoIDInstance(String repoIDString){
+        String lowerCaseRepoIDString = repoIDString.toLowerCase();
+        String[] repoIDComponents = lowerCaseRepoIDString.split("/");
+        if(repoIDComponents.length == 2){
+            RepoID repoID = new RepoID(lowerCaseRepoIDString, repoIDComponents[0], repoIDComponents[1]);
+            return Optional.of(repoID);
+        }
+        else{
+            return Optional.empty();
+        }
+    }
+
+    public String getRepoOwner() {
+        return repoOwner;
+    }
+
+    public String getRepoName() {
+        return repoName;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+
+        RepoID repoID = (RepoID) o;
+        return repoIDString.equals(repoID.repoIDString);
+
+    }
+
+    @Override
+    public int hashCode() {
+        return repoIDString.hashCode();
+    }
+
+    private RepoID(String lowerCaseRepoIDString, String repoOwner, String repoName){
+        this.repoIDString = lowerCaseRepoIDString;
+        this.repoOwner = repoOwner;
+        this.repoName = repoName;
+    }
```

Comment:
Be sure to take the subtleties of equals method into account e.g. http://stackoverflow.com/questions/27581/what-issues-should-be-considered-when-overriding-equals-and-hashcode-in-java